packages/material-ui-unstyled/src/utils/mergeEventHandlers.test.ts

```diff
@@ -0,0 +1,93 @@
+import { expect } from 'chai';
+import { spy } from 'sinon';
+import mergeEventHandlers from './mergeEventHandlers';
+
+describe('mergeEventHandlers', () => {
+  it('calls the provided event handlers', () => {
+    const callLog: string[] = [];
+
+    const ownHandlers = {
+      onClick: (_: any, next: () => void) => {
+        callLog.push('own');
+        next();
+      },
+    };
+
+    const externalHandlers = {
+      onClick: () => callLog.push('external'),
+    };
+
+    const mergedHandlers = mergeEventHandlers(ownHandlers, externalHandlers);
+    mergedHandlers.onClick(null);
+
+    expect(callLog).to.deep.equal(['own', 'external']);
+  });
+
+  it('calls the last provided handler if there is more than one available', () => {
+    const callLog: string[] = [];
+
+    const ownHandlers = {
+      onClick: (_: any, next: () => void) => {
+        callLog.push('own');
+        next();
```

Comment:
In case of the `useButton` (and, specifically, using it to implement `ButtonBase`) we need control over when the dev-supplied event handler gets called (see https://github.com/mui-org/material-ui/blob/next/packages/material-ui/src/ButtonBase/ButtonBase.js#L216 for example). That's why I introduced the callback parameter to the handlers internal to `useButton`.

Reply:
As pointed out in https://github.com/mui-org/material-ui/pull/27600#discussion_r690967884: Let's go without abstraction first to move this along.